src/chrome/startIdleChecking.js

```diff
@@ -0,0 +1,46 @@
+import { CONFIG } from '../constants';
+import { updateWorkerState } from '../actions';
+import listenStoreChanges from '../listenStoreChanges';
+
+const idlePeriod = 5 * 60;
+
+const notification = {
+  iconUrl: CONFIG.chrome.notificationIconUrl,
+  isClickable: true,
+  type: 'basic',
+  title: 'We noticed you were idle',
+  message: 'You seem to have been idle for a while, so we stopped ' +
+    'checking for work. Click here to start checking for work again.',
+};
+
+const workerIdle = 'workerIdle';
+
+const goIdle = (store, chrome) => {
+  store.dispatch(updateWorkerState('inactive'));
+  chrome.notifications.create(workerIdle, notification);
+  chrome.notifications.onClicked.addListener(notificationId => {
```

Comment:
this Event Listener should be loaded once.   During the lifecycle of this extension,  this will be loaded often and store.dispatch won't be called once but multiples of times.